File: 3239-minimum-number-of-flips-to-make-binary-grid-palindromic-i/3239-minimum-number-of-flips-to-make-binary-grid-palindromic-i.java
class Solution {
    public int minFlips(int[][] grid) {

        int ans1 = 0;

        for (int i = 0; i < grid.length; i++) {
            int j = 0;
            int k = grid[0].length - 1;
            while (j < k) {
                if (grid[i][j] != grid[i][k]) {
                    ans1++;
                }
                j++;
                k--;
            }
        }

        int ans2 = 0;

        for (int i = 0; i < grid[0].length; i++) {
            int j = 0;
            int k = grid.length - 1;
            while (j < k) {
                if (grid[j][i] != grid[k][i]) {
                    ans2++;
                }
                j++;
                k--;
            }
        }

        return Math.min(ans1, ans2);
    }
}

File: 3239-minimum-number-of-flips-to-make-binary-grid-palindromic-i/3239-minimum-number-of-flips-to-make-binary-grid-palindromic-i.cpp
class Solution {
public:
    int minFlips(vector<vector<int>>& grid) {
        int rcount = 0;
        for(int i = 0; i< grid.size(); i++){
            int a = 0; 
            int b = grid[0].size() -1;
            while(a < b){
                if(grid[i][a] != grid[i][b]){
                    rcount++;
                }
                a++, b--;
            }
        }

        // column palindrome 
        int cCount = 0;
        for(int j = 0; j< grid[0].size(); j++){
            int a = 0; 
            int b = grid.size() -1;
            while(a < b){
                if(grid[a][j] != grid[b][j]){
                    cCount++;
                }
                a++, b--;
            }
        }
        return min(rcount, cCount);
    }
};